services/media/service/src/ingest/handlers/upsert-localization-source-entity-finished-handler.ts

```diff
@@ -0,0 +1,182 @@
+  constructor(
+    private broker: Broker,
+    private loginPool: LoginPgPool,
+    config: Config,
+    overrides?: SubscriptionConfig,
+  ) {
+    super(
+      LocalizationServiceMultiTenantMessagingSettings
+        .UpsertLocalizationSourceEntityFinished.messageType,
+      ['INGESTS_EDIT', 'ADMIN'],
+      config,
+      overrides,
+    );
+  }
+
+  async onMessage(
+    content: UpsertLocalizationSourceEntityFinishedEvent,
+    message: MessageInfo,
+  ): Promise<void> {
+    const messageContext = message.envelope.message_context as Pick<
+      IngestMessageContext,
+      'ingestItemId'
+    >;
+    if (
+      !messageContext?.ingestItemId ||
+      content.service_id !== this.config.serviceId
+    ) {
+      // skipping message without ingest context or from different service
+      return;
+    }
+
+    const pgSettings = await this.getPgSettings(message);
+    const { ingestItem, localizationStep } = await transactionWithContext(
+      this.loginPool,
+      IsolationLevel.Serializable,
+      pgSettings,
+      async (ctx) => {
+        const ingestItem = await selectOne(
+          'ingest_items',
+          { id: messageContext.ingestItemId },
+          { columns: ['item'] },
+        ).run(ctx);
+        const localizationStep = await selectOne(
+          'ingest_item_steps',
+          {
+            ingest_item_id: messageContext.ingestItemId,
+            type: 'LOCALIZATIONS',
+          },
+          { columns: ['id'] },
+        ).run(ctx);
+        return { ingestItem, localizationStep };
+      },
+    );
+
+    if (!ingestItem || !localizationStep) {
+      throw new MosaicError({
+        message: `Unable to find existing Ingest Item with ID '${messageContext.ingestItemId}' or related Ingest Item Step with type 'LOCALIZATIONS'.`,
+        code: CommonErrors.IngestError.code,
+      });
+    }
+
+    const inputLocalizations = ingestItem.item.data
+      .localizations as IngestLocalization[];
+    const localizations: EntityLocalization[] = inputLocalizations.flatMap(
+      ({ language_tag, ...fields }) => ({
+        language_tag,
+        fields: Object.keys(fields).map((key) => ({
+          field_name: key,
+          field_value: fields[key] as string,
+          state: DEFAULT_LOCALIZATION_STATE,
+        })),
+      }),
+    );
+
+    const messageSettings =
+      LocalizationServiceMultiTenantMessagingSettings.LocalizeEntity;
+    const messagePayload: LocalizeEntityCommand = {
+      service_id: this.config.serviceId,
+      entity_type: content.entity_type,
+      entity_id: content.entity_id,
+      localizations,
+    };
+
+    const localizationMessageContext: IngestMessageContext = {
+      ingestItemId: messageContext.ingestItemId,
+      ingestItemStepId: localizationStep.id,
+    };
+
+    await this.broker.publish<LocalizeEntityCommand>(
+      UNKNOWN_AGGREGATE_ID,
```

Comment:
Maybe set the entity ID here as  `content.entity_id`? The aggregate type is "localization" so obviously it would not be the only key. But it would IMO still be the best thing to set there because it is not really unknown but rather just "not enough to exactly specify the entity without also using values from the payload".